attestation-agent/attester/src/utils.rs

```diff
@@ -13,3 +19,148 @@ pub fn pad<const T: usize>(input: &[u8]) -> [u8; T] {
     }
     output
 }
+
+/// This is a fixed eventlog header. If no CCEL is found, we will use this header ahead
+/// of aael eventlog.
+///
+/// The content of this HEADER is:
+///
+/// 1. EvNoAction entry
+/// 2. Digest sizes with sha384 -> 0x30
+///
+/// TODO: apply hash algorithm based on the underlying platform.
+/// Now the digests declared by CCEL might be different from the AAEL (defaults to sha384).
+/// This would cause the integrity check to fail.
+///
+/// The digest algorithm is determined by OVMF, and now tdx supports sha384. Let's extend it
+/// if RTMR and CCEL are brought for TEE except TDX.
+pub const EL_HEADER: [u8; 65] = [
+    0x00, 0x00, 0x00, 0x00, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x21, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x0C, 0x00, 0x30, 0x00,
+    0x00,
+];
+
+/// End flag for eventlog
+pub const EL_END_FLAG: [u8; 8] = [0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF];
+
+pub const DEFAULT_AAEL_PATH: &str = "/run/attestation-agent/eventlog";
+
+const CCEL_PATH: &str = "/sys/firmware/acpi/tables/data/CCEL";
+
+fn trim_ccel(mut ccel: Vec<u8>) -> Result<Vec<u8>> {
```

Comment:
it's too bad `/sys/firmware/acpi/tables/data/CCEL` is not like TPM `binary_bios_measurements` where we just get to read the TCG events and don't need any extra trimming for the data.

Reply:
yeah, it's OK to set `Closes: #569` in the cover letter.